cli/src/command_error.rs

```diff
@@ -554,16 +554,48 @@ jj currently does not support partial clones. To use jj with this repository, tr
                     err,
                     "Run `jj git remote rename` to give a different name.",
                 ),
-                GitPushError::RefInUnexpectedLocation(refs) => user_error_with_hint(
-                    format!(
-                        "Refusing to push a bookmark that unexpectedly moved on the remote. \
-                         Affected refs: {}",
-                        refs.join(", ")
-                    ),
-                    "Try fetching from the remote, then make the bookmark point to where you want \
-                     it to be, and push again.",
-                ),
-                GitPushError::RefUpdateRejected(_) => user_error(err),
+                GitPushError::FailedRefs {
+                    unexpected_locations,
+                    rejected_updates,
+                } => {
+                    let reference_errors = unexpected_locations
+                        .iter()
+                        .map(|(reference, reason)| {
+                            let reason = reason
+                                .as_ref()
+                                .map(|x| format!(" {x}"))
+                                .unwrap_or("".to_string());
+                            format!("{reference}: unexpectedly moved on the remote{reason}")
```

Comment:
The ` (…)` is part of [Git’s output format](https://github.com/git/git/blob/683c54c999c301c2cd6f715c411407c413b1d84e/transport.c#L630-L637) rather than from the remote. We should probably strip off `" ("` and `")"` when parsing rather than including it in our return value.